### PR TITLE
fix: misspecified types

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
- [ ] I have battle-tested on Overtaci (RMAPPS1279)
- [x] I have bumped the version as described in the wiki

## Notes for reviewers

I suspect these are leftovers from before we did code reviews @MartinBernstorff. However, if you were unaware (I believe you already know) `Optional[str] == Union[str, None]`.

### CODE REVIEW

File: src/psycopt2d/utils.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def drop_records_if_datediff_days_smaller_than(
         t2_col_name (str): _description_
         t1_col_name (str): _description_
         threshold_days (Union[float, int]): _description_
```
Comment (KennethEnevoldsen) [inline src/psycopt2d/utils.py:54]: Missing docstrings. I am not completely sure if this is correct.

```suggestion
        t2_col_name (str): Column name of a time column.
        t1_col_name (str): Column name of a time column.
        threshold_days (Union[float, int]): Threshold days.
```

File: src/psycopt2d/utils.py [hunk 2]
```diff
@@ -22,13 +22,13 @@ def flatten_nested_dict(
...
         parent_key (str): The parent key for the current dict, e.g. "level1" for the first iteration.
-        sep (str, optional): How to separate each level in the dict. Defaults to ".".
+        sep (str): How to separate each level in the dict. Defaults to ".".
 
     Returns:
         Dict: The flattened dict.
     """
 
-    items = []
+    items: List = []
```
Comment (MartinBernstorff) [inline src/psycopt2d/utils.py:31]: This is one of those cases where I think `darglint`/`mypy` is being overzealous.

**Review by MartinBernstorff — APPROVED** [review]

**Comment by MartinBernstorff** [comment]
How did you end up looking at this repo now, Kenneth? Now we have to fix failing imports ;-)

I was unaware about optional, so that's great to know! I'll have to adjust the docstrings a few places.

Feel free to merge when checks pass 👍

**Comment by KennethEnevoldsen** [comment]
I borrowed the flatten dict in another project and noticed som errors.

**Comment by MartinBernstorff** [comment]
Still failing. I'll look at this the next time I need T2D :-)

**Comment by KennethEnevoldsen** [comment]
It is sorry didn't mean to re-request a review. It works locally. I will rerun on a ucloud instance as well

**Comment by KennethEnevoldsen** [comment]
This install just fine on ucloud as well.

**Review by MartinBernstorff — COMMENTED** [review]
Just adding a review here to not get notifications.

**Comment by MartinBernstorff** [comment]
Well this uncovered a can of worms. Should be fixed by #194, but the commit history here is not great. I've re-done it in #195.

**Pull request closed without merging**